test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ModelBuilder/RecursiveComplexTypesTests.cs

```diff
@@ -0,0 +1,489 @@
+                    {
+                        Street = "123 Main Street",
+                        City = "Seattle",
+                        Country = "USA",
+                        PreviousAddress = new RecursiveComplexTypesTests_Address
+                        {
+                            Street = "111 West 8th Avenue",
+                            City = "Vancouver",
+                            Country = "Canada"
+                        }
+                    },
+                    HomeDirectory = new RecursiveComplexTypesTests_Directory
+                    {
+                        Name = string.Format("/users/customer{0}", i),
+                        Size = 2048,
+                        Files = Enumerable.Range(0, i).Select(j => new RecursiveComplexTypesTests_File
+                        {
+                            Name = string.Format("users/customer{0}/file{1}", i, j),
+                            Size = 1024 * j
+                        }).Concat(Enumerable.Range(0, 2).Select(j => new RecursiveComplexTypesTests_Directory
+                        {
+                            Name = string.Format("users/customer{0}/subdir{1}", i, j),
+                            Size = 1024 * j,
+                            Files = Enumerable.Range(0, j).Select(k => new RecursiveComplexTypesTests_File
+                            {
+                                Name = string.Format("users/customers{0}/subdir{1}/file{2}", i, j, k),
+                                Size = 1024 * k
+                            }).ToList()
+                        })).ToList()
+                    },
+                    CustomFields = new[]
+                    {
+                        new RecursiveComplexTypesTests_Field
+                        {
+                            Name = "LastLoginDate",
+                            DataType = "DateTime"
+                        },
+                        new RecursiveComplexTypesTests_Field
+                        {
+                            Name = "Referrals",
+                            DataType = "ComplexType",
+                            SubFields = new[]
+                            {
+                                new RecursiveComplexTypesTests_Field
+                                {
+                                    Name = "Name",
+                                    DataType = "String"
+                                },
+                                new RecursiveComplexTypesTests_Field
+                                {
+                                    Name = "ReferralBonus",
+                                    DataType = "Double"
+                                }
+                            }.ToList()
+                        }
+                    }.ToList()
+                }).ToList();
+            }
+
+            [EnableQuery]
+            public IQueryable<RecursiveComplexTypesTests_UserEntity> Get() => Users.AsQueryable();
```

Comment:
>=> [](start = 75, length = 2)

?

Reply:
This is an expression-bodied method. This has been in C# since version 6, introduced three years ago.

Reply:
Yes. I know the syntax. Just make sure our build agent are happy for the new syntax.

---
In reply to: [193254249](https://github.com/OData/webapi/pull/1458#discussion_r193254249) [](ancestors = 193254249)

Reply:
Thanks.

---
In reply to: [193579385](https://github.com/OData/webapi/pull/1458#discussion_r193579385) [](ancestors = 193579385)